mesonbuild/modules/pkgconfig.py

```diff
@@ -329,6 +330,27 @@ def _generate_pkgconfig_file(self, state, deps, subdirs, name, description,
                                  url, version, pcfile, conflicts, variables,
                                  unescaped_variables, uninstalled=False, dataonly=False):
         coredata = state.environment.get_coredata()
+        referenced_vars = {'prefix'}
+        optnames = [x.name for x in BUILTIN_DIR_OPTIONS.keys()]
+
+        if not dataonly:
+            # includedir is always implied, although libdir may not be
+            # needed for header-only libraries
+            referenced_vars |= {'includedir'}
+            if deps.pub_libs or deps.priv_libs:
+                referenced_vars |= {'libdir'}
+        # also automatically infer variables referenced in other variables
+        varnames = []
+        varstrings = []
+        for k, v in variables + unescaped_variables:
+            varnames.append(k)
+            varstrings.append(v)
+        for optname in optnames:
+            if optname not in varnames:
+                optvar = f'${{{optname}}}'
+                if any(x.startswith(optvar) for x in varstrings):
```

Comment:
Shouldn't it be `optvar in x` instead? The variable could be in the middle of the value, like `foo=${prefix}/${libdir}`.

Reply:
That would be broken, because you'd end up with for example:

 ```
prefix=/usr
libdir=${prefix}/lib
foo=${prefix}/${libdir}/foo
```

And then foo would expand to `/usr/usr/lib/foo`.

Any user doing that is better served by the pkgconfig module generating a broken pkg-config file that errors out, than by a pkg-config file that subtly returns the wrong answer.

Reply:
Good point, nevermind.